contracts/interchain-token/src/contract.rs

```diff
@@ -20,27 +46,171 @@ impl InterchainToken {
         );
 
         env.storage().instance().set(&DataKey::Initialized, &true);
+        env.storage().instance().set(&DataKey::TokenId, &token_id);
+        write_metadata(&env, token_meta_data);
 
-        env.storage().instance().set(&DataKey::Owner, &owner);
+        env.storage().instance().set(&DataKey::Admin, &admin);
+        env.storage()
+            .persistent()
+            .set(&DataKey::Minter(minter), &true);
+        env.storage()
+            .persistent()
+            .set(&DataKey::Minter(interchain_token_service.clone()), &true);
+
+        env.storage().instance().set(
+            &DataKey::InterchainTokenServiceAddress,
+            &interchain_token_service,
+        );
 
         Ok(())
     }
 
-    pub fn owner(env: &Env) -> Result<Address, ContractError> {
+    pub fn token_id(env: &Env) -> Address {
+        env.storage().instance().get(&DataKey::TokenId).unwrap()
+    }
+
+    pub fn interchain_token_service(env: &Env) -> Address {
         env.storage()
             .instance()
-            .get(&DataKey::Owner)
-            .ok_or(ContractError::NotInitialized)
+            .get(&DataKey::InterchainTokenServiceAddress)
+            .unwrap()
     }
 
-    pub fn transfer_ownership(env: Env, new_owner: Address) -> Result<(), ContractError> {
-        let owner: Address = Self::owner(&env)?;
-        owner.require_auth();
+    pub fn mint(env: Env, minter: Address, to: Address, amount: i128) -> Result<(), ContractError> {
+        minter.require_auth_for_args((&to, amount).into_val(&env));
```

Comment:
is this the recommended pattern? How is it different from just doing require_auth() for the invocation?

Reply:
isn't that implied by the fact that the minter had to authorize the invocation already (`to` and `amount` are all the arguments to this invocation).

Reply:
updated.